### PR TITLE
=str Resort the `OnNext` extractor order.

### PR DESCRIPTION
**Pull request by He-Pin**
Resort the cases for better performance

### CODE REVIEW

**Review by johanandren — COMMENTED** [review]
Seems reasonable, but, did you measure a difference?

**Comment by He-Pin** [comment]
> Seems reasonable, but, did you measure a difference?

Noop, but it is more consistency with other code, which OnNext comes in the first case.

File: akka-stream/src/main/scala/akka/stream/impl/FanIn.scala [hunk 1]
```diff
@@ -253,6 +251,8 @@ import akka.util.unused
           if (!receivedInput && isAllCompleted()) onCompleteWhenNoInput()
         case OnError(id, e) =>
           onError(id, e)
+        case OnSubscribe(id, subscription) =>
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/FanIn.scala:254]: OnNext is absolutely most common, others only happens once. I would place this before OnComplete, just to preserve the logical order that it happens before OnComplete

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**